Popup fix (#1573)
okta/okta-auth-js: README.md

});
```

> #### `token.getWithIDPPopup(options)`

Reviewer: Should we get into how this works from a technical perspective in the README?

1. The popup and main window can no longer communicate via [`window.postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage)
2. The main window can no longer detect if the popup window is closed

[`token.getWithIDPPopup`](#tokengetwithidppopupoptions) is designed for deployments which require the use of a popup window _and_ rely on external IDPs. This method can authenticate a user regardless of the IDP's [`Cross-Origin-Opener-Policy`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy), however it does come with some tradeoffs

##### Usage
```javascript
const { promise, cancel } = authClient.token.getWithIDPPopup({
  redirectUri: 'http://localhost:8080/popup/callback',
});
const { tokens } = await promise;
authClient.tokenManager.setTokens(tokens);
```

#### Tradeoffs

> 1. Since [`window.postMessage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) can no longer be replied on, the popup window's flow needs to end on the same origin as the application requesting tokens. After successful authentication, the popup window will be redirect to the provided `redirectUri`, which needs to call `authClient.handleIDPPopupRedirect()`. In Single Page Apps (with a router), it's recommended to host a dedicated route, but this logic can be perform on the main page as well.

Reviewer: > can no longer be replied on

typo: can no longer be relied upon

> but this logic can be perform on the

typo: but this logic can be performed on the

> NOTE: this will _not_ use the same value as `redirectUri` passed via the `OktaAuth` constructor

```javascript
// example implementation
authClient.handleIDPPopupRedirect();
window.close();     // recommended, closes the popup window
```

2. As mentioned above, the main window cannot detect when the popup window is closed. If a user manually closes the popup window before completing authentication, the resulting `promise` variable will still be `pending` (until a configurable timeout). A `cancel` method is provided to prevent awaiting for the promise to timeout, however this may still result in an awkward user experience.

>     1. It's important to provide a button on the page to invoke `cancel`. If the user closes the popup window, but `cancel` is never called, the `promise` will eventually timeout. This would obviously result in a poor UX.

Reviewer: Some wording here is a little bit conversational and awkward for docs:
```suggestion
    1. It's important to provide a button on the page to invoke `cancel`. If the user closes the popup window without calling `cancel`, the `promise` will eventually timeout. This results in a poor user experience and is should be avoided.
```

>     2. However, assuming a `cancel` button is available on the page; It's possible for a user to select the `cancel` action on the main window without closing the popup window. If the user _then_ completes the authentication flow in the popup window, this will _not_ result in tokens being issued to the main window (application). Creating another possibly awkward user experience.

Reviewer: ```suggestion
    2. However, assuming a `cancel` button is available on the page, it's possible for a user to select the `cancel` action on the main window without closing the popup window. If the user _then_ completes the authentication flow in the popup window, this will _not_ result in tokens being issued to the main window (application). This also results in a poor user experience and should be avoided.
```

> 

Reviewer: Might want to point at the example test app in here somewhere
